wcfsetup/install/files/lib/util/UnfurlUrlUtil.class.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+namespace wcf\util;
+
+use GuzzleHttp\Exception\BadResponseException;
+use GuzzleHttp\Exception\TooManyRedirectsException;
+use GuzzleHttp\Exception\TransferException;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\RequestOptions;
+use wcf\system\io\HttpFactory;
+
+/**
+ * Helper class to unfurl specific urls.
+ *
+ * @author      Joshua Ruesweg
+ * @copyright   2001-2021 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\Util
+ * @since       5.4
+ */
+final class UnfurlUrlUtil
```

Comment:
I dislike the classname `Url`, because we have already a class named `Url` which is used inside this class. What about `UnfurledUrl`?

Reply:
I recently attempted to somewhat avoid the [Smurf Naming Convention](https://devcards.io/smurf-naming-convention), because it tends to get overly long in many cases.

However I agree that the `Url` clash is unfortunate and that `Url` also is a super generic class name. I don't particularly like `UnfurledUrl`, though. How about `UrlResponse` or `UnfurlResponse`? These still should combine rather nicely with the proposed static `fetch()` method.